File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.java
class Solution {
    public int[] sortByBits(int[] arr) {

        Integer[] temp = Arrays.stream(arr).boxed().toArray(Integer[]::new);

        Arrays.sort(temp, (a, b) -> {
            int x = Integer.bitCount(a);
            int y = Integer.bitCount(b);
            if (x == y)
                return a - b;
            return x - y;
        });

        for (int i = 0; i < arr.length; i++)
            arr[i] = temp[i];

        return arr;
    }
}

File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.cpp
class Solution {
public:
    static int getSetBit(int n) {
        int c = 0;
        while (n) {
            n = n & (n - 1);
            c++;
        }
        return c;
    }
    vector<int> sortByBits(vector<int>& arr) {
        auto customComparator = [](int a, int b) {
            int x = getSetBit(a);
            int y = getSetBit(b);
            if (x == y)
                return a < b;
            return x < y;
        };
        sort(arr.begin(), arr.end(), customComparator);
        return arr;
    }
};